bears/vcs/git/GitCommitBear.py

```diff
@@ -24,4 +25,36 @@ def get_remotes():
 
     def get_head_commit(self):
         with change_directory(self.get_config_dir() or os.getcwd()):
+            command = self.check_github_pull_request_temporary_merge_commit()
+            if command:
+                return run_shell_command(command)
+
             return run_shell_command('git log -1 --pretty=%B')
+
+    def check_github_pull_request_temporary_merge_commit(self):
+        """
+        This detects the merge commit generated by GitHub
+        in a refs/pull/(\d+)/merge git remote reference.
+        Visit https://github.com/travis-ci/travis-ci/issues/8400
+        for more details.
+
+        :return: shell command to fetch the unmerged parent commit
```

Comment:
I think it might be better to use sentence like, "A git command (str) to fetch the unmerged parent commit if HEAD commit is a GitHub PR temporary merge commit, otherwise `None`".